datadog/resource_datadog_logs_custom_pipeline.go

```diff
@@ -417,61 +417,61 @@ func buildTerraformProcessors(ddProcessors []datadogV1.LogsProcessor) ([]map[str
-		logsDateRemapper := ddProcessor.LogsProcessorInterface.(*datadogV1.LogsDateRemapper)
-		tfProcessor = buildTerraformDateRemapper(logsDateRemapper)
-	case datadogV1.NewLogsMessageRemapperWithDefaults().GetType():
-		logsMessageRemapper := ddProcessor.LogsProcessorInterface.(*datadogV1.LogsMessageRemapper)
-		tfProcessor = buildTerraformMessageRemapper(logsMessageRemapper)
-	case datadogV1.NewLogsServiceRemapperWithDefaults().GetType():
-		logsServiceRemapper := ddProcessor.LogsProcessorInterface.(*datadogV1.LogsServiceRemapper)
-		tfProcessor = buildTerraformServiceRemapper(logsServiceRemapper)
-	case datadogV1.NewLogsStatusRemapperWithDefaults().GetType():
-		logsStatusRemapper := ddProcessor.LogsProcessorInterface.(*datadogV1.LogsStatusRemapper)
-		tfProcessor = buildTerraformStatusRemapper(logsStatusRemapper)
-	case datadogV1.NewLogsTraceRemapperWithDefaults().GetType():
-		logsTraceRemapper := ddProcessor.LogsProcessorInterface.(*datadogV1.LogsTraceRemapper)
-		tfProcessor = buildTerraformTraceRemapper(logsTraceRemapper)
-	case datadogV1.NewLogsGeoIPParserWithDefaults().GetType():
-		logsGeoIPParser := ddProcessor.LogsProcessorInterface.(*datadogV1.LogsGeoIPParser)
-		tfProcessor = buildTerraformGeoIPParser(logsGeoIPParser)
-	case datadogV1.NewLogsGrokParserWithDefaults().GetType():
-		logsGrokParser := ddProcessor.LogsProcessorInterface.(*datadogV1.LogsGrokParser)
-		tfProcessor = buildTerraformGrokParser(logsGrokParser)
-	case datadogV1.NewLogsLookupProcessorWithDefaults().GetType():
-		logsLookupProcessor := ddProcessor.LogsProcessorInterface.(*datadogV1.LogsLookupProcessor)
-		tfProcessor = buildTerraformLookupProcessor(logsLookupProcessor)
-	case datadogV1.NewLogsPipelineProcessorWithDefaults().GetType():
-		logsPipelineProcessor := ddProcessor.LogsProcessorInterface.(*datadogV1.LogsPipelineProcessor)
-		tfProcessor, err = buildTerraformNestedPipeline(logsPipelineProcessor)
-	case datadogV1.NewLogsStringBuilderProcessorWithDefaults().GetType():
-		logsStringBuilderProcessor := ddProcessor.LogsProcessorInterface.(*datadogV1.LogsStringBuilderProcessor)
-		tfProcessor = buildTerraformStringBuilderProcessor(logsStringBuilderProcessor)
-	case datadogV1.NewLogsURLParserWithDefaults().GetType():
-		logsURLParser := ddProcessor.LogsProcessorInterface.(*datadogV1.LogsURLParser)
-		tfProcessor = buildTerraformURLParser(logsURLParser)
-	case datadogV1.NewLogsUserAgentParserWithDefaults().GetType():
-		logsUserAgentParser := ddProcessor.LogsProcessorInterface.(*datadogV1.LogsUserAgentParser)
-		tfProcessor = buildTerraformUserAgentParser(logsUserAgentParser)
-	default:
-		err = fmt.Errorf("failed to support datadogV1 processor type, %s", ddProcessor.LogsProcessorInterface.GetType())
+	if ddProcessor.LogsArithmeticProcessor != nil {
+		tfProcessor = buildTerraformArithmeticProcessor(ddProcessor.LogsArithmeticProcessor)
+		processorType = string(datadogV1.LOGSARITHMETICPROCESSORTYPE_ARITHMETIC_PROCESSOR)
```

Comment:
Can you get it from `tfProcessorTypes[tfProcessor]`?

Reply:
Would this work? 🤔

Reply:
no, it wouldn't 😞  I missed the return type.